### PR TITLE
Initial merge

### PR DESCRIPTION
**Pull request by vitalych**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+First, you need to build S2E. Please refer to the S2E documentation for more details (http://s2e.systems).
+In all the commands below, do not forget to adapt the paths to fit your system.
+
+Second, recompile the sample binary with S2E support. This is necessary in order to write symbolic
+values to memory. At the moment, the hypervisor cannot write symbolic data to memory directly, so
+it must be done from the running binary instead.
+
+    cd sample
+    make clean
+    make EXTRA_CFLAGS="-DUSE_S2E -I/home/ubuntu/s2e/s2e/guest/common/include"
```
Comment (adrianherrera) [inline README.md]: Can we assume that the user is using s2e-env (make this clear in the README) and then this path can be collapsed to `$S2EDIR/source/s2e/guest/comm/include`?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    cd ..
+    
+    export S2E_CONFIG=sample/s2e-config.lua
+    export LD_PRELOAD=/home/ubuntu/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2e.so
```
Comment (adrianherrera) [inline README.md]: Same comment re s2e-env applies to this path

File: README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+This hypervisor is compatible with KVM extensions for symbolic execution provided by S2E.
+In other words, you can symbolically execute programs with this hypervisor.
+
+
+## Running with native KVM
+
+
+First, build the sample binary. This binary initializes some data at address `0x1000`.
+
+    cd sample
```
Comment (adrianherrera) [inline README.md]: Put inside

\`\`\` console
\`\`\`

tags

File: README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    make
+
+Run the compiled `sample.bin` binary in PyKVM. Make sure that `/dev/kvm` is available.
+
+    python -m pykvm.kvm sample/sample.bin
```
Comment (adrianherrera) [inline README.md]: Same backtick tags

File: README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+   can get symbolic data easier. You can view this library as an OS, or better yet the BIOS.
+   
+2. Extend PyKVM to load actual ELF/PE files into memory. Of course, they won't have any imports
+   or OS dependencies.
+
+3. Run PyKVM in GDB together with `libs2e`. The `gdb.ini` script contains the required
+   configuration for S2E. Please adapt the paths in there.
+
+
+    gdb --init-command gdb.ini --args python -m pykvm.kvm sample/sample.bin
```
Comment (adrianherrera) [inline README.md]: Backtick tags make things render nice :)

File: gdb.ini [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+#
+# Note: don't forget to update the paths
+
+handle SIGUSR2 noprint
+handle SIGUSR1 noprint
+set disassembly-flavor intel
+set print pretty on
+set environment S2E_CONFIG=sample/s2e-config.lua
+# set environment S2E_SHARED_DIR=
+set environment LD_PRELOAD=/home/vitaly/s2e/build/libs2e-release/x86_64-s2e-softmmu/libs2e.so
```
Comment (adrianherrera) [inline gdb.ini]: Ditch `/home/vitaly` (the user probably isn’t going to have that path :) and use `$S2EDIR` instead

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    version='1.0',
+    url='http://s2e.systems',
+    download_url='https://github.com/S2E/pykvm.git',
+    install_requires=[
+        'ioctl-opt',
+        'hexdump'
+    ],
+
+    test_requires=[
+        'mock',
```
Comment (adrianherrera) [inline setup.py]: Why is this required? Probably remove.

File: build.sh [hunk 8]
```diff
@@ -0,0 +1,23 @@
+#!/bin/sh
```
Comment (adrianherrera) [inline build.sh:1]: What is this script for? It doesn’t seem to be used/referenced anywhere

Comment (vitalych) [reply build.sh:1]: It's for building and uploading the package, makes my life easier.

**Review by adrianherrera — APPROVED** [review]
lgtm
